core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -297,23 +328,40 @@ public void moveBlock(long sessionId, long blockId, BlockStoreLocation oldLocati
       BlockStoreLocation newLocation)
           throws BlockDoesNotExistException, BlockAlreadyExistsException,
           InvalidWorkerStateException, WorkerOutOfSpaceException, IOException {
-    for (int i = 0; i < MAX_RETRIES + 1; i++) {
-      MoveBlockResult moveResult = moveBlockInternal(sessionId, blockId, oldLocation, newLocation);
-      if (moveResult.getSuccess()) {
-        synchronized (mBlockStoreEventListeners) {
-          for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-            listener.onMoveBlockByClient(sessionId, blockId, moveResult.getSrcLocation(),
-                moveResult.getDstLocation());
+    if (RESERVER_ENABLED) {
+      RetryPolicy retryPolicy = new TimeoutRetry(FREE_SPACE_TIMEOUT, EVICTION_INTERVAL_MS);
+      while (retryPolicy.attemptRetry()) {
+        MoveBlockResult result = moveBlockInternal(sessionId, blockId, oldLocation, newLocation);
+        if (result.getSuccess()) {
+          synchronized (mBlockStoreEventListeners) {
+            for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+              listener.onMoveBlockByClient(sessionId, blockId, result.getSrcLocation(),
+                  result.getDstLocation());
+            }
           }
+          return;
         }
-        return;
       }
-      if (i < MAX_RETRIES) {
-        freeSpaceInternal(sessionId, moveResult.getBlockSize(), newLocation);
+      throw new WorkerOutOfSpaceException(ExceptionMessage.NO_SPACE_FOR_BLOCK_MOVE_TIMEOUT,
+          newLocation, blockId, FREE_SPACE_TIMEOUT);
+    } else {
+      RetryPolicy retryPolicy = new CountingRetry(MAX_RETRIES);
+      while (retryPolicy.attemptRetry()) {
```

Comment:
In this file, are all the retry loops the same? They look the same, with the only difference being the retry policy, and whether or not we call `freeSpaceInternal`?

Reply:
The loops are the same, I plan to refactor it after we decide to go with either eviction strategy.